File: java/src/RMC.java
/**
 * @author Siôn Griffiths - sig2@aber.ac.uk
 *         Date: 09/03/14
 *         Time: 12:37
 */
public class RMC extends Sentence {

  public RMC(String[] input){
    data = input;

  }

  public GPSposition makeGPSposition(){
    int time = Integer.parseInt(data[1]);
    float lat = Float.parseFloat(data[3]);
    float lng = Float.parseFloat(data[5]);

    lat /= 100;
    lng /= 100;
    GPSposition gps = new GPSposition(time,lat, lng );



    return gps;
  }
}

File: cpp/RMC.cpp
/**
 * @file   RMC.cpp
 * @Author Sion Griffiths (sig2@aber.ac.uk)
 * @date   March 2014
 * @brief  Extracts GPS information from an NMEA RMC sentence  
 */

#include "RMC.h"
#include <string>
#include <vector>
#include <cstdlib>

/**
 * Constructs an RMC
 * @param stream pointer to current Stream
 * @param sentence vector of NMEA sentence tokens
 */
RMC::RMC(Stream *stream, vector<string> sentence) {
    makeRMCtime(stream, sentence);
}

RMC::RMC(const RMC& orig) {
}

RMC::~RMC() {
}

/**
 * Extracts latitude, longitude and elevation values from NMEA sentence.
 * Updates current Stream with these values
 * @param stream pointer to current Stream
 * @param sentence vector of NMEA sentence tokens
 */
void RMC::makeRMCposition(Stream* stream, vector<string> sentence) {
    lat = atof(sentence[3].c_str());
    lng = atof(sentence[5].c_str());

    if ((sentence[4].compare("S")) == 0) {
        lat *= -1;
    }
    if ((sentence[6].compare("W")) == 0) {
        lng *= -1;
    }



    stream->setCurrentLatLong(lat, lng);

}

/**
 * Extracts time values from NMEA sentence.
 * Updates current Stream with these values
 * @param stream pointer to current Stream
 * @param sentence vector of NMEA sentence tokens
 */
void RMC::makeRMCtime(Stream* stream, vector<string> sentence) {
    time = atoi(sentence[1].c_str());
    stream->setCurrentTime(time);
}